FinSearchUnified/Bundles/SearchBundleDBAL/QueryBuilderFactory.php

```diff
@@ -0,0 +1,193 @@
+    ) {
+        $this->connection = $connection;
+        $this->conditionHandlers = $container->get('shopware_searchdbal.condition_handlers');
+    }
+
+    /**
+     * @param Criteria $criteria
+     * @param ShopContextInterface $context
+     *
+     * @return QueryBuilder
+     * @throws \Exception
+     */
+    public function createQuery(Criteria $criteria, ShopContextInterface $context)
+    {
+        $query = $this->createQueryBuilder();
+
+        $this->prepareConditionHandlers($criteria);
+
+        $query->from('s_articles', 'product');
+
+        $query->leftJoin(
+            'product',
+            's_articles_details',
+            'mainDetail',
+            'mainDetail.id = product.main_detail_id'
+        );
+
+        $query->leftJoin(
+            'product',
+            's_articles_details',
+            'variant',
+            'variant.articleID = product.id AND variant.id != product.main_detail_id'
+        );
+
+        $this->addConditions($criteria, $query, $context);
+
+        return $query;
+    }
+
+    /**
+     * @param QueryBuilder $query
+     */
+    private function addSorting(QueryBuilder $query)
+    {
+        $query->addOrderBy('product.id', 'ASC');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createQueryBuilder()
+    {
+        return new QueryBuilder($this->connection);
+    }
+
+    /**
+     * @param Criteria $criteria
+     * @param ShopContextInterface $context
+     *
+     * @return QueryBuilder
+     * @throws \Exception
+     */
+    public function createProductQuery(Criteria $criteria, ShopContextInterface $context)
+    {
+        $query = $this->createQueryWithSorting($criteria, $context);
+
+        $select = $query->getQueryPart('select');
+
+        $query->select([
+            'SQL_CALC_FOUND_ROWS product.id AS __product_id',
+            'mainDetail.ordernumber AS __main_detail_number',
+            "GROUP_CONCAT(variant.ordernumber SEPARATOR ', ') AS __variant_numbers"
+        ]);
+
+        foreach ($select as $selection) {
+            $query->addSelect($selection);
+        }
+
+        $query->addGroupBy('product.id, mainDetail.ordernumber');
+
+        if ($criteria->getOffset()) {
+            $query->setFirstResult($criteria->getOffset());
+        }
+        if ($criteria->getLimit()) {
+            $query->setMaxResults($criteria->getLimit());
+        }
+
+        return $query;
+    }
+
+    /**
+     * @param Criteria $criteria
+     * @param QueryBuilder $query
+     * @param ShopContextInterface $context
+     *
+     * @throws \Exception
+     */
+    private function addConditions(Criteria $criteria, QueryBuilder $query, ShopContextInterface $context)
+    {
+        foreach ($criteria->getConditions() as $condition) {
+            $handler = $this->getConditionHandler($condition);
+            $handler->generateCondition($condition, $query, $context);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws \Exception
```

Comment:
Add an empty line before the `@throws`

Reply:
I had added that empty line, but I think the code styling added it back

Reply:
Oh okay. I thought you resolved the issue but then forgot to add the line. Okay, we will stick to the CS fixer's style next time.